[Leong Hong Fai] iP

leonghongfai opened the pull request:
# Duke

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/leonghongfai/ip.git)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😃 

And it is **FREE**!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

## Commands:

-  todo
- deadline
- event
- list
- done
- delete
- bye

## If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Duke {
    public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
    //other code...
}
```

Commit:
Add Gradle support

leonghongfai committed:
Added basic skeleton

leonghongfai committed:
Added level-2 stuff

leonghongfai committed:
Added task completion

leonghongfai committed:
Added different types of tasks

leonghongfai committed:
Testing done

leonghongfai committed:
Add exceptions handling

leonghongfai committed:
Add delete task functionality

leonghongfai committed:
Update tests to test delete functionality

leonghongfai committed:
Fix minor bugs

leonghongfai committed:
Add save feature

leonghongfai committed:
Add date feature for Event and Deadline

leonghongfai committed:
Merge branch 'branch-Level-7'

leonghongfai committed:
Merge branch 'branch-Level-8'

leonghongfai committed:
Make minor OOP changes

leonghongfai committed:
Revert "Make minor OOP changes"

This reverts commit 7ac3c2874223f36a6a66d791b6b9af624316a0bc.

leonghongfai committed:
Upgrade minor oop design

leonghongfai committed:
Upgrade oop structure

leonghongfai committed:
Complete minimal OOP requirement

leonghongfai committed:
Complete A-Package

leonghongfai committed:
Implement minimal JUnit testing

leonghongfai committed:
Create JAR file

leonghongfai committed:
Revamp of logic flow

leonghongfai committed:
Complete A-JavaDoc

leonghongfai committed:
Complete A-CodingStandard

leonghongfai committed:
Complete Level-9

leonghongfai committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Complete A-JavaDoc

leonghongfai committed:
Merge branch 'A-CodingStandard'

* A-CodingStandard:
  Complete A-CodingStandard

leonghongfai committed:
Merge branch 'Level-9'

* Level-9:
  Complete Level-9

leonghongfai committed:
Add missing javadocs

leonghongfai committed:
Edit incorrect test cases

leonghongfai commented:
# Duke

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/leonghongfai/ip.git)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😃 

And it is **FREE**!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

## Commands:

-  todo
- deadline
- event
- list
- done
- delete
- bye

## If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Duke {
    public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
    //other code...
}
```

leonghongfai committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into add-gradle-support

leonghongfai committed:
Add Gradle support

leonghongfai committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Add Gradle support
  Add Gradle support

leonghongfai committed:
Edit checkstyle flagged code

leonghongfai committed:
Merge branch 'A-CheckStyle'

NicolasCwy commented on README.md:
```diff
@@ -13,7 +13,7 @@ Prerequisites: JDK 11, update Intellij to the most recent version.
 1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
    In the same dialog, set the **Project language level** field to the `SDK default` option.
-3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
+3. After that, locate the `src/main/java/duke.java` file, right-click it, and choose `Run duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
```
Consider changing this to Duke.java as I thought you incorrectly named your file and made me check. Your Duke.java is capitalised correctly.

NicolasCwy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,157 @@
+     */
+    public String[] load() {
+        File textFile = new File("data/duke.txt");
+        Integer counter = 0;
```
I believe you can use int to access array indexes feel free to correct me if I'm wrong

NicolasCwy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,157 @@
+    /**
+     * Creates a folder and destination file if either of them do not exist.
+     */
+    public void makeFile() {
```
Good that you abstracted this out and named it as such!

NicolasCwy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,145 @@
+    public void start() {
+        System.out.println("Hello! I'm Duke\n");
+        System.out.println("What can I do for you?");
+        System.out.println("__________________________________________");
```
Is this meant to have less indentation than the later lines? Could consider abstracting this, in case you decide to change number of spaces later and saves you from copy-pasting!

NicolasCwy commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,69 @@
+ * Stores exceptions which generate a variety of error messages based on the error encountered during runtime.
+ */
+public class DukeException extends RuntimeException {
+    private final String INVALID_TODO = "Enter a valid todo in this format 'todo <task here>'";
```
Good capitalisation of constants here! 💯

NicolasCwy commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+ * Creates an ArrayList to store tasks and provides functionality to manipulate the tasks inside the ArrayList.
+ */
+public class TaskList {
+    private ArrayList<Task> store;
```
Perhaps rename this to something more descriptive like tasks?

NicolasCwy reviewed (commented):
Congratulations on making your first PR!🎉 Your code follows the standard conventions and I liked the use of abstraction for methods. Apart from a few nits in naming variables and maybe abstracting the text in the UI class LGTM!

leonghongfai committed:
Update gui

Bluntsord commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,69 @@
+    public DukeException(String type) {
+        this.type = type;
+        switch (type) {
+        case ("invalidToDo"):
```
Right way of indenting switch cases

Bluntsord commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+ *
+ * @author Leong Hong Fai
+ */
+
```
Nth much but may wanna remove this empty line for no reason

Bluntsord commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,140 @@
+                    store.add(temp);
+                    break;
+                case ("D"):
+                    Task temp1 = new Deadline(words[2], words[3]);
```
Please dont use numbers even in temporary variables

Bluntsord reviewed (commented):
Overall very clean coding standard and no code violations were made.Comments left were just minor mistakes for your discretion. Keep up the good work!

leonghongfai committed:
update fxml

leonghongfai committed:
Update GUI partially to reflect duke commands

leonghongfai committed:
Complete GUI implementation

leonghongfai committed:
Improve OOP design

leonghongfai committed:
Add javadocs

leonghongfai committed:
Fix "Find task" bug

leonghongfai committed:
No assertions present to catch logic flaws.

Assertions can help to define assumptions about the program state so
that the runtime can verify them.

Let's, add assertions liberally where suitable to verify that the
assumptions hold during runtime.

leonghongfai committed:
Refactor code

There are several deeply nested code that can make reading of code
difficult.

Refactoring the code improves readability drastically.

As a step towards better readability, let's refactor the code to remove
deeply nested blocks.

leonghongfai committed:
Merge pull request #1 from leonghongfai/A-Assertions

No assertions present to catch logic flaws.

leonghongfai committed:
Merge branch 'master' into A-CodeQuality

leonghongfai committed:
Merge pull request #2 from leonghongfai/A-CodeQuality

Refactor code

leonghongfai committed:
Set up CI

leonghongfai committed:
Add extension C-Undo

leonghongfai committed:
Add more features for extension

leonghongfai committed:
Make GUI improvements

leonghongfai committed:
Add screenshot

leonghongfai committed:
Make minor GUI improvements

leonghongfai committed:
Add User Guide

leonghongfai committed:
Make UG updates

leonghongfai committed:
Update UG

leonghongfai committed:
Fix logic flaw

leonghongfai committed:
Update screenshot

leonghongfai committed:
Fix code to fit coding standards

leonghongfai committed:
Change syntax to fit coding standards
